File: cpp/2294.cpp
class Solution {
public:
    int partitionArray(vector<int>& n, int k) {
        int s = n[0], e = n[0], p = 0, c = 0;
        for (int i : n) {
            s = min(s, i);
            e = max(e, i);
        }
        e = e - s + 1;
        vector<bool> f(e);
        for (int i : n)
            f[i - s] = true;
        for (int i = 0; i < e; i++)
            if (f[i] && i - p > k) {
                p = i;
                c++;
            }
        return c + 1;
    }
};

File: python3/2294.py
class Solution:
    def partitionArray(self, n: List[int], k: int) -> int:
        s = min(n)
        e = max(n) - s + 1
        p = 0
        c = 0
        f = [False] * e
        for i in n:
            f[i - s] = True
        for i in range(e):
            if f[i] and i - p > k:
                p = i
                c += 1
        return c + 1
